ArdalisRating/AllRatings/LifePolicy.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+
+namespace ArdalisRating.AllRatings
+{
+    public class LifePolicy : IRatingCheck
+    {
+        Printer printer = new();
+        public decimal Rate(Policy policy)
```

Comment:
Think about Dependency Injection here, and also this is a very long function!
You should think how to refactor this